Python/import.c

```diff
@@ -2221,15 +2223,70 @@ _imp_load_lazy_import_impl(PyLazyImport *lazy_import)  // was _imp_import_deferr
     return obj;
 }
 
+int
+_PyImport_LazyImportMatchEagerLoadedFilter(PyObject *name)
+{
+    PyInterpreterState *interp = _PyInterpreterState_GET();
+
+    // check if this module satisfied with `excluding` condition
+    // i.e. module name is in the excluded list or matches callback function
+    PyObject *filter = interp->eager_loaded_filter;
+
+    if (filter == NULL) {
+        return 0;
+    }
+
+    // match the list filter
+    if (PySet_Check(filter) && PySet_Contains(filter, name)) {
+        return 1;
+    }
+
+    // match the cb function filter
+    if (PyFunction_Check(filter) &&
+        PyObject_CallObject(filter, PyTuple_Pack(1, name)) == Py_True)
+    {
+        return 1;
+    }
```

Comment:
I think the logic here needs to be more flexible to account for any callable object (not just PyFunctionObject) and any iterable. But this should really all be handled already in `set_lazy_imports`. As I suggested above to use two different fields on the interpreter state, this function would be simplified to just "check if there's a callback, if so call it, otherwise check for membership in the set."